server/src/main/java/org/opensearch/cluster/metadata/QueryGroup.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the QueryGroup schema
+ * {
+ *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga",
+ *              "jvm": 0.4,
+ *              "resiliency_mode": "enforced",
+ *              "name": "analytics",
+ *              "updatedAt": 4513232415
+ * }
+ */
+@ExperimentalApi
+public class QueryGroup extends AbstractDiffable<QueryGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final String _id;
+    private final QueryGroupMode resiliencyMode;
+    // It is an epoch in millis
+    private final long updatedAtInMillis;
+    private final Map<String, Object> resourceLimits;
+
+    // list of resources that are allowed to be present in the QueryGroup schema
+    public static final List<String> ALLOWED_RESOURCES = List.of("jvm", "cpu");
+
+    public QueryGroup(String name, String _id, QueryGroupMode resiliencyMode, Map<String, Object> resourceLimits, long updatedAt) {
```

Comment:
Ideally we should have another constructor that generates the random uuid and updatedAt internally. Probably, we need that in future